src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/ExecuteReconfigureTasksControllerTest.java

```diff
@@ -0,0 +1,256 @@
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.wataskmanagementapi.config.SecurityConfiguration.SERVICE_AUTHORIZATION;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskOperationName.EXECUTE_RECONFIGURE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskOperationName.MARK_TO_RECONFIGURE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.utils.ServiceMocks.SERVICE_AUTHORIZATION_TOKEN;
+
+@SuppressWarnings("checkstyle:LineLength")
+class ExecuteReconfigureTasksControllerTest extends SpringBootIntegrationBaseTest {
+
+    public static final String A_TASK_NAME = "follow Up Overdue Reasons For Appeal";
+    public static final String A_TASK_TYPE = "followUpOverdueReasonsForAppeal";
+    public static final String SOME_ASSIGNEE = "someAssignee";
+    OffsetDateTime createdDate = OffsetDateTime.now();
+    OffsetDateTime dueDate = createdDate.plusDays(1);
+
+    private static final String ENDPOINT_BEING_TESTED = "/task/operation";
+
+    @MockBean
+    private ClientAccessControlService clientAccessControlService;
+    @MockBean
+    private ConfigureTaskService configureTaskService;
+    @MockBean
+    private TaskAutoAssignmentService taskAutoAssignmentService;
+
+    @MockBean
+    private CaseConfigurationProviderService caseConfigurationProviderService;
+
+    @SpyBean
+    private CFTTaskDatabaseService cftTaskDatabaseService;
+
+    private String taskId;
+    private TaskResource testTaskResource;
+
+    @BeforeEach
+    void setUp() {
+        taskId = UUID.randomUUID().toString();
+        when(clientAccessControlService.hasExclusiveAccess(SERVICE_AUTHORIZATION_TOKEN))
+            .thenReturn(true);
+
+        lenient().when(caseConfigurationProviderService.evaluateConfigurationDmn(anyString(),
+            any())).thenReturn(List.of(
+            new ConfigurationDmnEvaluationResponse(
+                CamundaValue.stringValue("caseName"),
+                CamundaValue.stringValue("Value"),
+                CamundaValue.booleanValue(true)
+            )
+        ));
+
+        testTaskResource = new TaskResource(taskId, "someTaskName", "someTaskType", CFTTaskState.ASSIGNED, "caseId101", dueDate);
+        testTaskResource.setCreated(OffsetDateTime.now());
+
+        when(configureTaskService.configureCFTTask(any(TaskResource.class), any(TaskToConfigure.class)))
+            .thenReturn(testTaskResource);
+
+        when(taskAutoAssignmentService.reAutoAssignCFTTask(any(TaskResource.class)))
+            .thenReturn(testTaskResource);
+
+    }
+
+    @Test
+    void should_execute_reconfigure_for_reconfigure_request_time() throws Exception {
```

Comment:
As this is doing both operations, should we make it clear via the test name? should_successfully_mark_to_configure_and_then_execute_reconfigure_on_task

Reply:
done